pynamodb/attributes.py

```diff
@@ -884,7 +972,7 @@ def _get_class_for_serialize(value):
     if value is None:
         return NullAttribute()
     if isinstance(value, MapAttribute):
-        return type(value)()
```

Comment:
Any idea on why we might've been doing this before? I can't think of one, since `.serialize()` shouldn't mutate the object

Reply:
No idea -- this hasn't changed since it was introduced with List and Map support:
http://github.com/pynamodb/PynamoDB/pull/175

Reply:
I'm going to move this particular change into a different PR:
https://github.com/pynamodb/PynamoDB/pull/839